export_snowflake/__init__.py

```diff
@@ -522,26 +528,126 @@ def flush_records(stream: str,
     db_sync.delete_from_stage(stream, s3_key)
 
 
+def transfer_data_in_parallel(s3_csv_lists, config, o, file_format_type):
+    parallel_transfer_start_time = time.time()
+    
+    # make the decision of how many processes to use later
+    pool = multiprocessing.Pool(4, initialize_s3_client_db_sync(config, o, file_format_type))
+    pool.map(download_file, s3_csv_lists)
+    pool.close()
+    pool.join()
+    
+    parallel_transfer_end_time = time.time()
+    stream = config["stream"]
+    LOGGER.info(f"Start time for transfering {stream} as {parallel_transfer_start_time} and end time as {parallel_transfer_end_time}")
+
+
+def download_file(job):
+    bucket, key, filename, stream = job
+
+    s3_client.download_file(bucket, key, filename)
```

Comment:
How slow will this operation (s3 -> snowflake) will be if it does not use local disk at all?

Reply:
Directly loading file from S3 to Snowflake will result in extra charge when Snowflake is reading data from external cloud service provider. With reading from S3 to local firstly, won't result in extra fees compared to what we are doing right now.

Reply:
Might want to double check this. I think Snowflake only charges data transfer fee for data egress, but not ingress. I don't think there should be any other extra compute/storage costs compared to original code.  
https://docs.snowflake.com/en/user-guide/cost-understanding-data-transfer

Reply:
Right, Snowflake doesn't seem to charge for ingress feels. No extra fees compared to original code. 
We still can't bypass the token expiration for Snowflake with direct s3 usage.

Reply:
I was referring to not using local disk to download whole file and then only uploading it to S3. We read from S3 ourselves and write to Snowflake ourselves but not use local disk as intermediate step. Whats the perf difference if we do that?